src/components/app/data/hooks/useEnterpriseLearner.test.jsx

```diff
@@ -19,37 +20,94 @@ const mockEnterpriseCustomer = enterpriseCustomerFactory();
   enterpriseFeatures: {},
   staffEnterpriseCustomer: null,
+  shouldUpdateActiveEnterpriseCustomerUser: false,
 };
 
+const mockBFFDashboardData = {
+  enterpriseCustomer: {
+    ...mockEnterpriseCustomer,
+    isBFFEnabled: true,
+  },
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
+};
+
+const mockExpectedEnterpriseLearner = (isMatchedBFFRoute) => (isMatchedBFFRoute
+  ? {
+    enterpriseCustomer: mockBFFDashboardData.enterpriseCustomer,
+    allLinkedEnterpriseCustomerUsers: mockBFFDashboardData.allLinkedEnterpriseCustomerUsers,
+    enterpriseFeatures: mockBFFDashboardData.enterpriseFeatures,
+  }
+  : mockEnterpriseLearnerData);
+
 describe('useEnterpriseLearner', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ routes = null, children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={[routes]}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
   beforeEach(() => {
     jest.clearAllMocks();
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
     useParams.mockReturnValue({ enterpriseSlug: mockEnterpriseCustomer.slug });
   });
-  it('should handle resolved value correctly', async () => {
-    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseLearner(), { wrapper: Wrapper });
-    await waitForNextUpdate();
 
-    expect(result.current).toEqual(
-      expect.objectContaining({
-        data: mockEnterpriseLearnerData,
-        isLoading: false,
-        isFetching: false,
-      }),
+  it.each(generateTestPermutations({
+    isMatchedBFFRoute: [false, true],
+    hasCustomSelect: [false, true],
+  }))('should return enterprise learners correctly (%s)', async ({
+    isMatchedBFFRoute,
+    hasCustomSelect,
+  }) => {
+    const mockSelect = jest.fn(data => data.transformed);
+    const enterpriseLearnerHookArgs = hasCustomSelect ? { select: mockSelect } : {};
+    const { result, waitForNextUpdate } = renderHook(
+      () => (useEnterpriseLearner(enterpriseLearnerHookArgs)),
+      {
+        wrapper: ({ children }) => Wrapper({
+          routes: isMatchedBFFRoute ? '/test-enterprise' : 'test-enterprise/search',
```

Comment:
nit: a bit odd to pass a singular `route` to an attr for (plural) `routes`. Maybe the `Wrapper` component should expect its prop to be an array, such that `initialEntries={routes}`. 

As is, `initialEntries={[routes]}` largely implies an array inside an array, at face value.